### ISSUE TITLE
Is it possible to create an in memory graph database for integration testing purposes?

### ISSUE THREAD

**Issue by zifnab87** [datastax/java-driver-dse#43]
In Titan it was possible to have an in memory database that was used for integration purposes. Is it or will it be shortly possible to do that with DataStax DSE Graph?

Thanks,
Michail

**Comment by tolbertam** [comment]
I don't think it is currently possible, for our integration tests we've been using [CCM](https://github.com/pcmanus/ccm), which works with DSE:

Base class:

https://github.com/datastax/java-driver-dse/blob/1.x/src/test/java/com/datastax/driver/dse/graph/CCMGraphTestsSupport.java

Example test:

https://github.com/datastax/java-driver-dse/blob/1.x/src/test/java/com/datastax/driver/dse/graph/GraphIntegrationTest.java

**Comment by zifnab87** [comment]
@tolbertam Thanks! So basically if I pass the "fake" DseSession from CCMGraphTestsSupport to my code through DI, I can test if it works properly assuming that I setup the graph the way I want before the test and remove any changes after the tests- right? 

It seems possible..

**Comment by olim7t** [comment]
It's not a fake session, CCM runs actual Cassandra/DSE instances locally, and the tests connect to it like they would to a remote cluster. It uses a temporary data directory that gets cleaned up at the end.
I don't think our test infrastructure is easily reusable right now, `CCMGraphTestsSupport` is not deployed to Maven central, and we use an internal fork of CCM customized for DSE. We're considering publishing our test infrastructure as a standalone project to make things easier.

On a side note, I'm going to close GitHub issues on this project in favor of our existing JIRA bugtracker. I'll add a link to it in the README file.
